dali/operators/reader/nvdecoder/dynlink_nvcuvid.cc

```diff
@@ -110,57 +114,60 @@ static CUresult LOAD_LIBRARY(DLLDRIVER *pInstance)
 
 CUresult cuvidInit(unsigned int Flags)
 {
-    DLLDRIVER DriverLib;
-
-    CHECKED_CALL(LOAD_LIBRARY(&DriverLib));
+  CHECKED_CALL(LOAD_LIBRARY(&DriverLib));
 
-    // fetch all function pointers
-    GET_PROC(cuvidCreateVideoSource);
-    GET_PROC(cuvidCreateVideoSourceW);
-    GET_PROC(cuvidDestroyVideoSource);
-    GET_PROC(cuvidSetVideoSourceState);
-    GET_PROC(cuvidGetVideoSourceState);
-    GET_PROC(cuvidGetSourceVideoFormat);
-    GET_PROC(cuvidGetSourceAudioFormat);
+  // fetch all function pointers
+  GET_PROC(cuvidCreateVideoSource);
+  GET_PROC(cuvidCreateVideoSourceW);
+  GET_PROC(cuvidDestroyVideoSource);
+  GET_PROC(cuvidSetVideoSourceState);
+  GET_PROC(cuvidGetVideoSourceState);
+  GET_PROC(cuvidGetSourceVideoFormat);
+  GET_PROC(cuvidGetSourceAudioFormat);
 
-    GET_PROC(cuvidCreateVideoParser);
-    GET_PROC(cuvidParseVideoData);
-    GET_PROC(cuvidDestroyVideoParser);
+  GET_PROC(cuvidCreateVideoParser);
+  GET_PROC(cuvidParseVideoData);
+  GET_PROC(cuvidDestroyVideoParser);
 
 
-    GET_PROC(cuvidGetDecoderCaps);
-    GET_PROC(cuvidCreateDecoder);
-    GET_PROC(cuvidDestroyDecoder);
-    GET_PROC(cuvidDecodePicture);
-    GET_PROC_OPTIONAL(cuvidGetDecodeStatus);
-    GET_PROC_OPTIONAL(cuvidReconfigureDecoder);
+  GET_PROC(cuvidGetDecoderCaps);
+  GET_PROC(cuvidCreateDecoder);
+  GET_PROC(cuvidDestroyDecoder);
+  GET_PROC(cuvidDecodePicture);
+  GET_PROC_OPTIONAL(cuvidGetDecodeStatus);
+  GET_PROC_OPTIONAL(cuvidReconfigureDecoder);
 
 #if !defined(__CUVID_DEVPTR64) || defined(__CUVID_INTERNAL)
-    GET_PROC(cuvidMapVideoFrame);
-    GET_PROC(cuvidUnmapVideoFrame);
+  GET_PROC(cuvidMapVideoFrame);
+  GET_PROC(cuvidUnmapVideoFrame);
 #endif
 
 #if defined(_WIN64) || defined(__LP64__) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
-    GET_PROC(cuvidMapVideoFrame64);
-    GET_PROC(cuvidUnmapVideoFrame64);
+  GET_PROC(cuvidMapVideoFrame64);
+  GET_PROC(cuvidUnmapVideoFrame64);
 #endif
-    GET_PROC(cuvidCtxLockCreate);
-    GET_PROC(cuvidCtxLockDestroy);
-    GET_PROC(cuvidCtxLock);
-    GET_PROC(cuvidCtxUnlock);
+  GET_PROC(cuvidCtxLockCreate);
+  GET_PROC(cuvidCtxLockDestroy);
+  GET_PROC(cuvidCtxLock);
+  GET_PROC(cuvidCtxUnlock);
 
-    return CUDA_SUCCESS;
+  return CUDA_SUCCESS;
 }
 
 bool cuvidInitChecked(unsigned int Flags) {
-    static std::mutex m;
-    static bool initialized = false;
+  if (initialized)
+    return true;
 
-    if (initialized)
-        return true;
+  std::lock_guard<std::mutex> lock(m);
+  static CUresult res = cuvidInit(Flags);
```

Comment:
this doesn't need to be static, does it?

Reply:
Done